arch/wasm/bits/alltypes.h.in

```diff
@@ -0,0 +1,33 @@
+#ifdef __wasm32__
+#define _Addr int
+#define _Int64 long long
+#define _Reg int
+
+#elif defined __wasm64__
+#define _Addr long
+#define _Int64 long long
+#define _Reg long
+
+#endif
+
+TYPEDEF __builtin_va_list va_list;
+TYPEDEF __builtin_va_list __isoc_va_list;
+
+#ifndef __cplusplus
+TYPEDEF __WCHAR_TYPE__ wchar_t;
+#endif
+TYPEDEF __WINT_TYPE__ wint_t;
+
+TYPEDEF float float_t;
+TYPEDEF double double_t;
+
+TYPEDEF long time_t;
```

Comment:
`long` is 32-bit on wasm32, so using it for `time_t` is susceptible to the [2038 bug](https://en.wikipedia.org/wiki/Year_2038_problem). Since wasm has a full set of 64-bit operations, I suggest making `time_t` 64-bit.

Reply:
Yes, what was I thinking!?! The ABI is supposed to match the Linux x32 ABI, this must have snuck in from the x86 one by mistake.

I'll do a diff between arch/x32 and arch/wasm and make sure everything matches where possible.

(By comparing the syscall numbers from all the different architectures, I've deduced that Emscripten is currently trying to emulate the x86 ABI in its syscall interface, or at least seems to have copied the definitions from x86 most extensively. I think moving to the x32 definitions is a good move at this point.)